yt_idefix/fields.py

```diff
@@ -47,3 +62,266 @@ def setup_fluid_fields(self):
+            )
+            self.alias(
+                ("gas", "Temperature"),
+                (self.ds._dataset_type, "Temperature"),
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "temperature"),
+                (self.ds._dataset_type, "Temperature"),
+                units=unit_system["temperature"],
+            )
+        elif (self.ds._dataset_type, "temperature") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "temperature"),
+                sampling_type="cell",
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "Temperature"),
+                (self.ds._dataset_type, "temperature"),
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "temperature"),
+                (self.ds._dataset_type, "temperature"),
+                units=unit_system["temperature"],
+            )
+        else:
+
+            def _temperature(field, data):
+                return (
+                    data.ds.mu
+                    * (mh / kboltz)
+                    * (data[("gas", "pressure")] / data[("gas", "density")])
+                )
+
+            self.add_field(
+                ("gas", "Temperature"),
+                sampling_type="cell",
+                function=_temperature,
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "temperature"),
+                ("gas", "Temperature"),
+                units=unit_system["temperature"],
+            )
+
+        if (self.ds._dataset_type, "mach") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "mach"),
+                sampling_type="cell",
+                units="",
+            )
+            self.alias(
+                ("gas", "Mach"),
+                (self.ds._dataset_type, "mach"),
+                units="",
+            )
+            self.alias(
+                ("gas", "mach"),
+                (self.ds._dataset_type, "mach"),
+                units="",
+            )
+        else:
+
+            def _mach(field, data):
+                return np.sqrt(
+                    data[(self.ds._dataset_type, "vx1")] ** 2
+                    + data[(self.ds._dataset_type, "vx2")] ** 2
+                    + data[(self.ds._dataset_type, "vx3")] ** 2
+                ) / np.sqrt(
+                    data.ds.gamma
+                    * (data[("gas", "pressure")] / data[("gas", "density")])
+                )
+
+            self.add_field(
+                ("gas", "Mach"),
+                sampling_type="cell",
+                function=_mach,
+                units="",
+            )
+            self.alias(
+                ("gas", "mach"),
+                ("gas", "Mach"),
+                units="",
+            )
+
```

Comment:
the mach number is defined for free by yt if you define the `("gas", "sound_speed")` field here instead

Reply:
Implemented `("gas", "sound_speed")` and removed `mach`. I wasn't aware of this yt feature before.